### ISSUE TITLE
Add CI for Linting and Code Coverage

### ISSUE THREAD

**Issue by vhespanha** [d4r1us-drk/clido#11]
(I promise this is my last issue today.)

## Description

I'm primarily opening this issue so it can be better tracked, since I'm already working on it for the past day or so. I also think it should be added to the milestone since this will make for a better DX for new contributors.

I decided to switch linters for the project to cover more cases and implement GitHub Actions for it. In this process, I found a lot of linting warnings that should probably be addressed before we can enforce them on the repo. Most of them will be fixed in the following PR, but others will need a little more thinking and big changes, which will be better suited for separate PRs or even issues (like [#10](https://github.com/d4r1us-drk/clido/issues/10)).

My strategy will be to:
- Fix the simpler linting warnings in the following PR.
- Address other warnings through their respective issues.
- Implement testing for the whole repo (help wanted, I'm terrible with tests).
- Finally, implement CI for the new linter and to check code coverage by tests.

## Benefits

- **Automated Linting**: Automatically check for code errors and formatting issues.
- **Code Coverage**: Measure the percentage of code covered by tests to identify untested areas.
- **Consistency**: Ensure that all code changes meet our quality standards before merging.
- **Better DX**: Improve the developer experience for new contributors.

## Tasks

(@d4r1us-drk, feel free to edit this to add or remove something, or to assign people to any task)

- [x] Switch to a new linter that covers more cases.
- [x] Fix simpler linting warnings in the next PR.
- [ ] Address more complex linting warnings through separate PRs or issues.
- [ ] Implement testing for the entire repo.
- [ ] Set up a GitHub Actions workflow for linting.
- [ ] Set up a GitHub Actions workflow for code coverage.

**Comment by dynnian** [comment]
> Implement testing for the whole repo (help wanted, I'm terrible with tests).

I am too lmao

**Comment by dynnian** [comment]
I will try to get more contributors and help on Mastodon and asking friends

**Comment by vhespanha** [comment]
Another task completed! [#14](https://github.com/d4r1us-drk/clido/pull/14/)
